### ISSUE TITLE
Attach Current Date property

### ISSUE THREAD

**Issue by PrzemyslawSwiderski** [JetBrains/gradle-changelog-plugin#9]
Hi,
it is a great plugin but it would be great if You could add some property to an extension in order to attach date to a release version, so instead of:
```
changelog {
    headerFormat = "[{0}] - {1}"
    headerArguments = listOf(version, SimpleDateFormat("yyyy-MM-dd").format(Date()))
}
```
we can write something like this:
```
changelog {
   attachDate = true // default = false
}
```
It would simplify code and let to avoid some SimpleDateFormat and Date imports in build files ;)
Once again thank You for this plugin and please let me know if this change is possible :)

**Comment by hsz** [comment]
Thanks, @PrzemyslawSwiderski!

I agree with you that the requirement of using the SDF makes everything a bit complicated.
If we'd go with your `attachDate` property - how would it be possible to configure the date format (i.e., set it as `dd-MM-yyyy`) or its position in the header string (place it after `-` or inside `()`, ...)?

What may be worth adding instead, can be a helper function, like `closure` one to generate the date in the given format, like:

```
import org.jetbrains.changelog.date

changelog {
    headerFormat = "[{0}] - {1}"
    headerArguments = listOf(version, date("yyyy-MM-dd"))
}
```

What do you think?

**Comment by PrzemyslawSwiderski** [comment]
Yes, this helper function would be definitely an improvement.

However, I am also thinking about some other property like `headerType` or `headerMode` which would specify how header should look like. 
Possible values of this property could be as follows:
* `default` - version without date (default)
* `withDate` - version with a date format `yyyy-mm-dd` like the one in example: https://keepachangelog.com/en/1.0.0/
* `custom` - header can be fully customized by `headerFormat` and `headerArguments` properties.
Any other values should produce some exception "Illegal headerType in changelog plugin, possible values: default, withDate or custom.".

Please let me know if it sounds good :)

**Comment by hsz** [comment]
I think that it overcomplicates the header configuration.

Let's go through the available options:
- `default` is already covered if you omit the `headerFormat` and `headerArguments` properties,
- `custom` is now equal to the case when you specify the above properties. If `headerType` would be here, you'll need extra work to make it work because of setting `headerType = "custom"`,
- `withDate` - will force you to use a unified header format. If you'd like to alter it, you'll end up with `custom` mode case anyway and specifying the `SimpleDateFormat` as an argument.

As a side effect, the header configuration will be more complicated from the implementation and documentation point of view.

I've reviewed the current state of the format/arguments properties, and it was possible to simplify this process to handle generating the header text with closure function:

```groovy
import org.jetbrains.changelog.date

changelog {
    header = { "[${project.version}] - ${date()}" }
}
```

By default, it will still be set to the `{ "[${project.version}]" }`, however, now adding the date to the header will be much simpler.

**Comment by hsz** [comment]
Released with `v0.5.0`.

**Comment by PrzemyslawSwiderski** [comment]
Great, thanks.
